### PR TITLE
Finish functional TODOs 

### LINKED ISSUE #7560: Complete misc TODOs in `TransactionProcessor` and `ProxyEvmAccount`

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #7560 
 - Finishes (afaik) the functional TODOs remaining in the `TransactionProcessor` code path needed for `Erc721OperationsTest` added [here](https://github.com/hashgraph/hedera-services/pull/7588) to pass given a `HandleContextScope` implementation.

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/bni/Dispatch.java [hunk 1]
```diff
@@ -254,7 +254,7 @@ ResponseCodeEnum transferWithReceiverSigCheck(
      * @param amount                 the amount to charge
      * @param itemizeStoragePayments whether to itemize storage payments in the record
      */
-    ResponseCodeEnum chargeStorageRent(long contractNumber, long amount, boolean itemizeStoragePayments);
+    void chargeStorageRent(long contractNumber, long amount, boolean itemizeStoragePayments);
```
Comment (lukelee-sl) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/bni/Dispatch.java:257]: why did this change?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java [hunk 2]
```diff
@@ -420,4 +431,8 @@ private com.hedera.hapi.node.state.token.Account validatedAccount(final long num
         }
         return account;
     }
+
+    private UInt256 valueOrZero(@Nullable final SlotValue slotValue) {
+        return (slotValue == null) ? UInt256.ZERO : pbjToTuweniUInt256(slotValue.value());
```
Comment (lukelee-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java:436]: Danno mentioned that the tuweni libraries are soon to be not supported.  Nothing to do for now but something to keep in mind for later.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/bni/Dispatch.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/bni/Dispatch.java:257]: Given the `HandleException`-based idiom already used by the `BaseProxyWorldUpdater.commit()` steps, throwing on an error here seemed more natural.

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Pull request merged**